subsys/bluetooth/controller/hci/hci.c

```diff
@@ -3624,33 +3630,83 @@ int hci_acl_handle(struct net_buf *buf, struct net_buf **evt)
 #endif /* CONFIG_BT_CONN */
 
 #if CONFIG_BT_CTLR_DUP_FILTER_LEN > 0
-static inline bool dup_found(struct pdu_adv *adv)
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+static void store_adi(int i, struct pdu_adv_adi *adi)
+{
+	if (adi) {
+		memcpy(&dup_filter[i].adi, adi, sizeof(*adi));
+	} else {
+		memset(&dup_filter[i].adi, 0, sizeof(*adi));
+	}
+}
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+
+static bool dup_found(uint8_t adv_type, uint8_t addr_type, uint8_t *addr,
+		      uint8_t adv_mode, struct pdu_adv_adi *adi,
+		      uint8_t data_status)
 {
 	/* check for duplicate filtering */
 	if (dup_count >= 0) {
 		int i;
 
 		for (i = 0; i < dup_count; i++) {
```

Comment:
I think it would be good to refactor the body of this loop into a separate function (returning something like "duplicate"/"updated"/"not_found") to help readability and keep indentation levels low.

Reply:
ok. will give a try.